pkg/admission/certwriter/fs.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certwriter
+
+import (
+	"strings"
+
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/admission/certgenerator"
+)
+
+const (
+	// FSCertProvisionAnnotationKeyPrefix should be used in an annotation in the following format:
+	// fs.certprovisioner.kubernetes.io/<webhook-name>: path/to/certs/
+	// the webhook cert manager library will provision the certificate for the webhook by
+	// storing it under the specified path.
+	// format: local.certprovisioner.kubernetes.io/webhookName: path/to/certs/
+	FSCertProvisionAnnotationKeyPrefix = "fs.certprovisioner.kubernetes.io/"
+)
+
+// FSCertWriterProvider deals with writing to the local filesystem.
+type FSCertWriterProvider struct {
+	CertGenerator certgenerator.CertGenerator
+}
+
+var _ CertWriterProvider = &FSCertWriterProvider{}
+
+// Provide creates a new CertWriter and initialized with the passed-in webhookConfig.
+func (s *FSCertWriterProvider) Provide(webhookConfig runtime.Object) (CertWriter, error) {
+	fsWebhookMap := map[string]*webhookAndPath{}
+
+	accessor, err := meta.Accessor(webhookConfig)
+	if err != nil {
+		return nil, err
+	}
+	annotations := accessor.GetAnnotations()
+	if annotations == nil {
+		return nil, nil
+	}
+
+	// Parse the annotations to extract info
+	for k, v := range annotations {
+		if strings.HasPrefix(k, FSCertProvisionAnnotationKeyPrefix) {
```

Comment:
Provide an example of the annotation here in a comment so it is easier to make sense of